lib/runtime/wasmer/imports.go

```diff
@@ -1093,9 +1120,36 @@ func ext_default_child_storage_storage_kill_version_1(context unsafe.Pointer, ch
 }
 
 //export ext_default_child_storage_storage_kill_version_2
-func ext_default_child_storage_storage_kill_version_2(context unsafe.Pointer, a, b C.int64_t) C.int32_t {
+func ext_default_child_storage_storage_kill_version_2(context unsafe.Pointer, childStorageKeySpan, _ C.int64_t) C.int32_t {
 	logger.Debug("[ext_default_child_storage_storage_kill_version_2] executing...")
-	logger.Warn("[ext_default_child_storage_storage_kill_version_2] unimplemented")
+	logger.Warn("[ext_default_child_storage_storage_kill_version_2] somewhat unimplemented")
+	// TODO: need to use `limit` parameter
+
+	instanceContext := wasm.IntoInstanceContext(context)
+	ctx := instanceContext.Data().(*runtime.Context)
+	storage := ctx.Storage
+
+	childStorageKey := asMemorySlice(instanceContext, childStorageKeySpan)
+	storage.DeleteChild(childStorageKey)
+
+	// note: this function always returns `KillStorageResult::AllRemoved`, which is 0
+	return 0
+}
+
+//export ext_default_child_storage_storage_kill_version_3
+func ext_default_child_storage_storage_kill_version_3(context unsafe.Pointer, childStorageKeySpan, _ C.int64_t) C.int64_t {
+	logger.Debug("[ext_default_child_storage_storage_kill_version_3] executing...")
+	logger.Warn("[ext_default_child_storage_storage_kill_version_2] somewhat unimplemented")
```

Comment:
i guess this log should be `ext_default_child_storage_storage_kill_version_3`

Reply:
updated